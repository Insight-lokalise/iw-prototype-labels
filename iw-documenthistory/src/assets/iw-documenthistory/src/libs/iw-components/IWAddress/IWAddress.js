import PropTypes from 'prop-types'
import React from 'react'

/**
 * A wrapper for a read-only address section.
 */
export function IWAddress({ address, className }) {
  const { address1, address2, city, state, zipCode, countryId } = address

  return (
    <p className={className}>
      {!!address1 && address1}
      <br />
      {!!address2 && (
        <span>
          {address2}
          <br />
        </span>
      )}
      {!!city && `${city}, `}
      {!!state && state} {!!zipCode && zipCode}
      <br />
      {!!countryId && countryId}
    </p>
  )
}

IWAddress.propTypes = {
  address: PropTypes.shape({
    address1: PropTypes.string,
    address2: PropTypes.string,
    city: PropTypes.string,
    state: PropTypes.string,
    zipCode: PropTypes.string,
    countryId: PropTypes.string,
  }),
}
