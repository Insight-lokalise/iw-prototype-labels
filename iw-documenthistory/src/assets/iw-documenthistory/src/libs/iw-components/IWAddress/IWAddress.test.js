import React from 'react'
import { cleanup, render, screen , within} from '@testing-library/react'
import { IWAddress } from './IWAddress'

let wrapper
const address = {
  address1: 'address1',
  address2: 'address2',
  city: 'city',
  state: 'state',
  zipCode: 'zipCode',
  countryId: 'countryId',
}

describe('IWAddress', () => {

  afterEach(() => {
    cleanup()
  })

  it('renders with full address', () => {
    const { asFragment } = render(<IWAddress address={address} className="rendered" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders w/o address1', () => {
    const { asFragment } = render(<IWAddress address={{...address, address1: undefined}} className="rendered" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders w/o address2', () => {
    const { asFragment } = render(<IWAddress address={{...address, address2: undefined}} className="rendered" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders w/o city', () => {
    const { asFragment } = render(<IWAddress address={{...address, city: undefined}} className="rendered" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders w/o state', () => {
    const { asFragment } = render(<IWAddress address={{...address, state: undefined}} className="rendered" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders w/o zipCode', () => {
    const { asFragment } = render(<IWAddress address={{...address, zipCode: undefined}} className="rendered" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders w/o countryId', () => {
    const { asFragment } = render(<IWAddress address={{...address, countryId: undefined}} className="rendered" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
