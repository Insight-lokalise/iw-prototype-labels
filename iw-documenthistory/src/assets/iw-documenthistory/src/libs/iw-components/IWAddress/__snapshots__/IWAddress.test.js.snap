// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IWAddress renders w/o address1 1`] = `
<DocumentFragment>
  <p
    class="rendered"
  >
    <br />
    <span>
      address2
      <br />
    </span>
    city, state zipCode
    <br />
    countryId
  </p>
</DocumentFragment>
`;

exports[`IWAddress renders w/o address2 1`] = `
<DocumentFragment>
  <p
    class="rendered"
  >
    address1
    <br />
    city, state zipCode
    <br />
    countryId
  </p>
</DocumentFragment>
`;

exports[`IWAddress renders w/o city 1`] = `
<DocumentFragment>
  <p
    class="rendered"
  >
    address1
    <br />
    <span>
      address2
      <br />
    </span>
    state zipCode
    <br />
    countryId
  </p>
</DocumentFragment>
`;

exports[`IWAddress renders w/o countryId 1`] = `
<DocumentFragment>
  <p
    class="rendered"
  >
    address1
    <br />
    <span>
      address2
      <br />
    </span>
    city, state zipCode
    <br />
  </p>
</DocumentFragment>
`;

exports[`IWAddress renders w/o state 1`] = `
<DocumentFragment>
  <p
    class="rendered"
  >
    address1
    <br />
    <span>
      address2
      <br />
    </span>
    city,  zipCode
    <br />
    countryId
  </p>
</DocumentFragment>
`;

exports[`IWAddress renders w/o zipCode 1`] = `
<DocumentFragment>
  <p
    class="rendered"
  >
    address1
    <br />
    <span>
      address2
      <br />
    </span>
    city, state 
    <br />
    countryId
  </p>
</DocumentFragment>
`;

exports[`IWAddress renders with full address 1`] = `
<DocumentFragment>
  <p
    class="rendered"
  >
    address1
    <br />
    <span>
      address2
      <br />
    </span>
    city, state zipCode
    <br />
    countryId
  </p>
</DocumentFragment>
`;
